File: Ex78_ViewBinding/app/src/main/java/com/example/ex78_viewbinding/MyAdapter.java
package com.example.ex78_viewbinding;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.ex78_viewbinding.databinding.RecyclerItemBinding;

import java.util.ArrayList;

public class MyAdapter extends RecyclerView.Adapter<MyAdapter.VH> {
    RecyclerItemBinding binding;
    Context context;
    ArrayList<ItemVO> items;

    public MyAdapter(Context context, ArrayList<ItemVO> items) {

        this.context = context;
        this.items = items;
    }

    @NonNull
    @Override
    public VH onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
//        binding = RecyclerItemBinding.inflate(LayoutInflater.from(context),parent,false);
//        return new VH(binding.getRoot());

        return new VH(LayoutInflater.from(context).inflate(R.layout.recycler_item,parent,false));
    }

    @Override
    public void onBindViewHolder(@NonNull VH holder, int position) {
        ItemVO itemVO = items.get(position);
        holder.binding.iv.setImageResource(itemVO.imgResId);
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    class VH extends RecyclerView.ViewHolder{

        // 20_ recycler_item.xml 과 연결되는 바인딩 클래스 참조변수
        RecyclerItemBinding binding;

        public VH(@NonNull View itemView) {
            super(itemView);
            // 21_ 이미 만든 뷰객체 안에 있는 자식뷰들과 연결하는 바인딩 객체를 만들자.
            binding = RecyclerItemBinding.bind(itemView);
        }
    }
}

File: Ex93_KotlinOpenApiNaverSearch/app/src/main/java/com/example/ex93_kotlinopenapinaversearch/MyAdapter.kt
package com.example.ex93_kotlinopenapinaversearch

import android.content.Context
import android.content.Intent
import android.net.Uri
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.text.HtmlCompat
import androidx.recyclerview.widget.RecyclerView.Adapter
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.bumptech.glide.Glide
import com.example.ex93_kotlinopenapinaversearch.databinding.RecyclerItemBinding

class MyAdapter(var context:Context,var items:MutableList<ShoppingItem>) : Adapter<MyAdapter.VH>() {
    inner class VH(var binding: RecyclerItemBinding) : ViewHolder(binding.root)
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): VH = VH(RecyclerItemBinding.inflate(LayoutInflater.from(context),parent,false))
    override fun getItemCount(): Int = items.size
    override fun onBindViewHolder(holder: VH, position: Int) {
        var item:ShoppingItem = items[position]
        
        // 16_ 제목 글씨 중에 html 태그문이 포함되어 있어서 지저분.. 이를 제거 하자
        var title:String = HtmlCompat.fromHtml(item.title,HtmlCompat.FROM_HTML_MODE_COMPACT).toString()
        holder.binding.tvTitle.text = title

        holder.binding.tvLowPrice.text = "${item.lprice} 원"
        holder.binding.tvMallName.text = item.mallName
        Glide.with(context).load(item.image).into(holder.binding.iv)

        holder.binding.root.setOnClickListener {
            // 17_ 디바이스의 인터넷 앱을 실행해보자.
            val intent = Intent(Intent.ACTION_VIEW)
            intent.data = Uri.parse(item.link)
            context.startActivity(intent)
        }
    }
}